scripts/opportunity-provider-limo/index.ts

```diff
@@ -0,0 +1,206 @@
+        symbol: opportunity.token2.symbol,
+      },
+      randomizeSides: opportunity.randomizeSides,
+      minAmountNotional: opportunity.minAmountNotional,
+      maxAmountNotional: opportunity.maxAmountNotional,
+    };
+  });
+}
+
+const decimals: Record<string, number> = {};
+const prices: Record<string, number> = {};
+
+async function getDecimals(
+  connection: Connection,
+  token: Token
+): Promise<number> {
+  const index = token.mint.toBase58();
+  if (decimals[index] === undefined) {
+    decimals[index] = await getMintDecimals(connection, token.mint);
+  }
+
+  return decimals[index];
+}
+
+async function getPrice(connection: Connection, token: Token): Promise<number> {
+  const index = token.symbol;
+  if (prices[index] === undefined) {
+    const url = `https://api.binance.com/api/v3/ticker/price?symbol=${token.symbol}USDT`;
+    const response = await fetch(url);
+    const data = await response.json();
+    const price = parseFloat(data.price);
+    if (isNaN(price)) {
+      throw new Error(`Invalid price: ${data.price}`);
+    }
+    prices[index] = price;
+  }
+
+  const mintDecimals = await getDecimals(connection, token);
+  return prices[index] / Math.pow(10, mintDecimals);
+}
+
+async function createOpportunities(
+  skExecutor: Keypair,
+  limoClient: limo.LimoClient,
+  opportunitiesPath: string,
+  count: number,
+  edge: number
+) {
+  const opportunities = loadOpportunities(opportunitiesPath);
+  for (let i = 0; i < opportunities.length; i++) {
+    const opportunity = opportunities[i];
+    for (let j = 0; j < count; j++) {
+      let inputToken = opportunity.token1;
+      let outputToken = opportunity.token2;
+      if (opportunity.randomizeSides) {
+        if (Math.random() > 0.5) {
+          inputToken = opportunity.token2;
+          outputToken = opportunity.token1;
+        }
+      }
+
+      const priceInput = await getPrice(limoClient.getConnection(), inputToken);
+      const priceOutput = await getPrice(
+        limoClient.getConnection(),
+        outputToken
+      );
+
+      const notional =
+        Math.random() *
+          (opportunity.maxAmountNotional - opportunity.minAmountNotional) +
+        opportunity.minAmountNotional;
+
+      const amountInput = (notional * (1 + edge / 10_000)) / priceInput;
+      const amountOutput = notional / priceOutput;
+
+      console.log("Creating opportunity:");
+      const decimalsInput = await getDecimals(
+        limoClient.getConnection(),
+        inputToken
+      );
+      const decimalsOutput = await getDecimals(
+        limoClient.getConnection(),
+        outputToken
+      );
+      console.log(
+        `Input: ${inputToken.symbol}, ${
+          amountInput / Math.pow(10, decimalsInput)
+        }`
+      );
+      console.log(
+        `Output: ${outputToken.symbol}, ${
+          amountOutput / Math.pow(10, decimalsOutput)
+        }`
+      );
+
+      const signature = (
+        await limoClient.createOrderGeneric(
```

Comment:
Maybe just create the limoClient here and use another variable to create the connection at the beginning of this function. It seems unnatural that we just get the connection from the client for other purposes.